### ISSUE TITLE
Will it trigger the next round of job if the last one has not completed?

### ISSUE THREAD

**Issue by kvzn** [mdsherry/clokwerk#32]
When we use something like `scheduler.every(1.seconds()).run(move || {})`, if the job is heavy and can't complete at 1 second, when the next second comes,  the last job has not finished, will it trigger a new round executing? (As I have tested, then answer looks like no). Thank you for official answer!

**Comment by mdsherry** [comment]
No, it will wait until the start of the next second to run again. The timing for the next run is computed after the task is complete. Even for less frequent tasks, you have a long(er) running task you should probably offload its execution into a separate thread so that you don't delay any other tasks. That is, given
```rust
scheduler.every(1.hour()).run(move || takes_five_minutes());
scheduler.every(1.second()).run(move || almost_instant());
```
executing `takes_five_minutes` will prevent `almost_instant` from running for five minutes of every hour.

Since I don't know the nature of the task, I don't know if it's acceptable for multiple copies of it to be running concurrently. If concurrent execution is fine, then you could just create a new thread/use a thread pool for each task. If you want the task executions to be serialized, but starting late is better than waiting, you could have the task executed by Clokwerk push details onto a work queue (e.g. [mpsc](https://doc.rust-lang.org/std/sync/mpsc/) or similar) which is read from a separate long-running thread. (In that case, do be wary of the queue growing indefinitely; you might want to use a [`SyncSender`](https://doc.rust-lang.org/std/sync/mpsc/struct.SyncSender.html) with a limit of 1, and make use of [`try_send`](https://doc.rust-lang.org/std/sync/mpsc/struct.SyncSender.html#method.try_send) to only push a new job if one isn't already pending.)

tl;dr: the logic behind `watch_thread` boils down to:
```
while not exiting:
   for each task:
      if the task's scheduled next execution time is in the past:
        run the task
        reschedule the task
   sleep a bit
```
